src/history/hs_rec.c

```diff
@@ -1072,19 +902,19 @@ __hs_delete_key_from_pos(WT_SESSION_IMPL *session, WT_CURSOR *hs_cursor, uint32_
          * The cursor will also be positioned at the start of the range that we wish to start
          * inserting.
          */
-        WT_WITHOUT_DHANDLE(session,
-          ret = __wt_open_cursor(session, WT_HS_URI, NULL, open_cursor_cfg, &insert_cursor));
+        WT_WITHOUT_DHANDLE(session, ret = __wt_curhs_open(session, NULL, &insert_cursor));
+        F_SET(insert_cursor, WT_CURSTD_HS_READ_COMMITTED);
```

Comment:
Here the same

Reply:
We are calling `search_near()` on the cursor here so we have to use this flag.

Reply:
I only see we do `WT_ERR(insert_cursor->insert(insert_cursor));` on the cursor. There is a search near call inside the insert implementation but that is on the underlying file cursor so should be fine.